politeiawww/util/proposal.go

```diff
@@ -0,0 +1,55 @@
+package util
+
+import (
+	"bufio"
+	"errors"
+	"io"
+	"strings"
+
+	www "github.com/decred/politeia/politeiawww/api/v1"
+	"github.com/kennygrant/sanitize"
+)
+
+const (
+	// IndexFile contains the file name of the index file
+	IndexFile = "index.md"
+)
+
+var (
+	// ErrIndexFileNotFound indicates that the index file was not found in the
+	// proposal files.
+	ErrIndexFileNotFound = errors.New("index.md file was not found")
+)
+
+// ProposalName returns the proposal name
+func ProposalName(files []www.File) (string, error) {
+	var (
+		fileFound bool
+		payload   string
+		name      string
+	)
+
+	// get index file payload
+	for _, f := range files {
+		if f.Name == IndexFile {
+			payload = f.Payload
+			fileFound = true
+			break
+		}
+	}
+	if !fileFound {
+		return "", ErrIndexFileNotFound
+	}
+
+	// @rgeraldes - used reader instead of scanner
+	// due to the size of the input (scanner > token too long)
+	// get the first line from the payload
+	reader := bufio.NewReader(strings.NewReader(payload))
+	name, err := reader.ReadString('\n')
```

Comment:
The file payload is in base64 so it must be decoded before you can extract the name.